### ISSUE TITLE
[FairMQ] Channel registry

### ISSUE THREAD

**Issue by dennisklein** [FairRootGroup/FairRoot#572]
- [x] Provide `RegisterChannelEndpoint(std::string channelname, int min_num_subchannels = 1, int max_num_subchannels = 1)` api through which the user code can declare which channels it expects during the `INITIALIZING DEVICE` device state.
- [x] Provide a command line option `--print-channels` which dumps all the registered channels and then exits.
- [x] ~~Add a check to the device which verifies a match between configured and registered channels.~~

**Comment by dennisklein** [comment]
@ktf Can you pls give an example how you would imagine the `--print-channel` to look like?

**Comment by ktf** [comment]
I think I would be happy with a flat list, e.g., if I do:

```C++
RegisterChannel("input", 2);
RegisterChannel("output");
RegisterChannel("debug");
```

I would expect that:

```bash
MyDevice --print-channels
```

returns:

```bash
input:1
output:2
debug:1
```

**Comment by rbx** [comment]
This is now available in the following form:
```C++
// override RegisterChannelEndpoints, that will be called by the framework
void YourDevice::RegisterChannelEndpoints()
{
    // provide channel name, minimum and maximum number of subchannels
    RegisterChannelEndpoint("channelA", 1, 10000);
    RegisterChannelEndpoint("channelB", 1, 1);
}
```
Starting this device with `--print-channels` will produce following output and exit the device:
```
channelB:1:1
channelA:1:10000
```

This functionality is optional - the device developer can register channels or not. Also, we will not do the check to verify match between registered and configured channels (at least for now), because there can be devices where channels names are not known in advance before the configuration takes place (e.g. FairMQMultiplier has configurable channel names at runtime).

**Comment by rbx** [comment]
Implemented in #592 and documented with #626.
